fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -3028,11 +3072,15 @@ public Bundle doHistory(MultivaluedMap<String, String> queryParameters, String r
             }
 
             responseBuilder.lastModified(com.ibm.fhir.model.type.Instant.of(changeRecord.getChangeTstamp().atZone(UTC)));
+            responseBuilder.location(Url.of(changeRecord.getResourceTypeName() + "/" + changeRecord.getLogicalId() + "/_history/" + changeRecord.getVersionId()));
 
+            // Per the R4 spec, the fullUrl should not contain _history/:vid
             Entry.Builder entryBuilder = Entry.builder();
-            entryBuilder.fullUrl(Url.of(changeRecord.getResourceTypeName() + "/" + changeRecord.getLogicalId() + "/_history/" + changeRecord.getVersionId()));
+            entryBuilder.id(Long.toString(changeRecord.getChangeId()));
+            entryBuilder.fullUrl(Url.of(changeRecord.getResourceTypeName() + "/" + changeRecord.getLogicalId()));
             entryBuilder.request(requestBuilder.build());
             entryBuilder.response(responseBuilder.build());
+            entryBuilder.resource(resource); // remember resource may be null if returnPreference=minimal
```

Comment:
this says `returnPreference=minimal` which matched my proposal in the issue, but in the docs we say `returnPreference=none`.
based on the logic above the two should work the same (since it only checks `returnPreference=representation`), but I think we should settle on one or the other to avoid any confusion.

Reply:
ok, i left a suggestion on the doc to say return=minimal ...assuming we make that update then this is good-to-go